src/GA4ServiceProvider.php

```diff
@@ -21,14 +22,31 @@ public function configurePackage(Package $package): void
             ->hasRoute('web');
     }
 
-    public function registeringPackage()
+    public function registeringPackage(): void
     {
         $this->app->bind('ga4', function () {
-            return new GA4MeasurementProtocol();
+            if (config('google-analytics-4-measurement-protocol.measurement_id') === null
+                || config('google-analytics-4-measurement-protocol.api_secret') === null
+            ) {
+                throw new Exception('Please set .env variables for Google Analytics 4 Measurement Protocol as per the readme file first.');
+            }
```

Comment:
Does this mean that someone will face this exception for any request just after installing the package until they set the .env vars?

Reply:
This called only on resolving 'ga4'. Same as in the constructor.